Fixed AI-assisted sql_query generation and made has_valid_schema compatible with older Spark versions (#995)
databrickslabs/dqx: docs/dqx/docs/dev/contributing.mdx

<Admonition type="warning" title="Special Cases">
* Avoid using backticks around object names in docstrings (e.g., \`arg1\`), as this can cause issues when rendering API documentation. Instead, use italics (e.g., \*arg1\*) to emphasize object names.
* Double curly braces are not allowed in the description. Mask them with backslashes, e.g.: `{{`.

> * If you need a code example in a docstring, use triple backticks. The following shows the literal text that must appear inside the docstring:

Reviewer: ```suggestion
* If you want to add a code example to a docstring, use triple backticks. The following shows the literal text that must appear inside the docstring:
```
